### PR TITLE
Get list of enjoy

### LINKED ISSUE #55: Get list of activity | event
Constraint => Select dates that come after the start date for events|activities, no need to display events on earlier dates

### PR DESCRIPTION
**Pull request by Nokomi6**
Get all item of enjoy in a particular city (params need to send)

Closes #55

### CODE REVIEW

File: server/controllers/enjoy.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	"server/graphql"
+	"server/utils"
+
+	"github.com/gin-gonic/gin"
+)
+
+type EnjoyParams struct {
+	City string `json:"city" binding:"required"`
+	Page int    `json:"page" binding:"required,gte=0"`
+}
```
Comment (MelvinCou) [inline server/controllers/enjoy.go:15]: Greater or equal 1 for pagination
Use form instead of json. Swagger generates documentation with small case, So this will be a problem for front client.
```suggestion
	City string `form:"city" binding:"required"`
	Page int    `form:"page" binding:"required,gte=1"`
}
```

File: server/controllers/enjoy.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+// Enjoy godoc
+//
+//	@Summary		Enjoy
+//	@Description	Get events|activities from our DATATourisme API
+//	@Tags			graphql
+//	@Accept			json
+//	@Produce		json
+//	@Param			enjoyQueryParams	query		EnjoyParams	true	"enjoyQueryParams"
+//	@Success		200					{object}	graphql.GetEnjoyResponse
+//	@Failure		500					{object}	graphql.Error
```
Comment (MelvinCou) [inline server/controllers/enjoy.go:26]: Add bad request

File: server/test/integration/drink_test.go [hunk 3]
```diff
@@ -3,33 +3,46 @@ package integration
...
 
+	"github.com/gin-gonic/gin"
 	"github.com/gin-gonic/gin/binding"
 	"github.com/stretchr/testify/assert"
 )
 
 const (
-	endpoint = "/api/drink"
+	endpointDrink = "/api/drink"
+	endpointEnjoy = "/api/enjoy"
```
Comment (MelvinCou) [inline server/test/integration/drink_test.go:18]: This may be a bad practice to set consts that are not related all in one. Setting  `endpointEnjoy` in `enjoy_test.go` doesn't work ?

File: server/test/integration/enjoy_test.go [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+	r.ServeHTTP(w, req)
+
+	// Check the HTTP status code is BadRequest
+	assert.Equal(t, http.StatusBadRequest, w.Code)
+}
+
+func TestGetEnjoyFromParis(t *testing.T) {
+	var resp graphql.GetEnjoyResponse
+	w := httptest.NewRecorder()
+	req, _ := http.NewRequest("GET", endpointEnjoy+"?City=Paris&Page=1", http.NoBody)
```
Comment (MelvinCou) [inline server/test/integration/enjoy_test.go:30]: You may need to change the query to small case

**Review by MelvinCou — CHANGES_REQUESTED** [review]

File: server/test/integration/enjoy_test.go (hunk 4 above)
Comment (Nokomi6) [reply server/test/integration/enjoy_test.go:30]: Can you explain more what you want ?

File: server/controllers/enjoy.go (hunk 1 above)
Comment (Nokomi6) [reply server/controllers/enjoy.go:15]: I fix it in another branch

File: server/test/integration/drink_test.go (hunk 3 above)
Comment (Nokomi6) [reply server/test/integration/drink_test.go:18]: I fix it in another branch

File: server/test/integration/enjoy_test.go (hunk 4 above)
Comment (MelvinCou) [reply server/test/integration/enjoy_test.go:30]: Change City => city
Page => page

Unit test may not pass if you don't chage this.

**Review by MelvinCou — APPROVED** [review]
